### ISSUE TITLE
AnimeBytes not working for a long time 

### ISSUE THREAD

**Issue by AliShaikh94** [Jackett/Jackett#945]
Just updated to the latest as I saw AnimeBytes mentioned in the release notes, but every time I try to connect I get: 

Jackett.Utils.Clients.WebClientByteResult ---> System.ArgumentNullException: Array cannot be null.                                  
Parameter name: bytes     

Full log: 

> Source value:                                                                                                                       
Jackett.Utils.Clients.WebClientByteResult ---> System.ArgumentNullException: Array cannot be null.                                  
Parameter name: bytes                                                                                                               
  at System.Text.Encoding.GetString (System.Byte[] bytes) <0x80625a9e0 + 0x0008a> in <filename unknown>:0                           
  at Jackett.JackettModule.<Load>m__2 (Jackett.Utils.Clients.WebClientByteResult be, Jackett.Utils.Clients.WebClientStringResult str
) <0x80c7d6f00 + 0x00030> in <filename unknown>:0                                                                                   
  at AutoMapper.Internal.MappingExpression`2+<>c__DisplayClass57_0[TSource,TDestination].<AfterMap>b__0 (System.Object src, System.O
bject dest) <0x80c7d6e60 + 0x00094> in <filename unknown>:0                                                                         
  at AutoMapper.TypeMap.<get_AfterMap>b__40_0 (System.Object src, System.Object dest) <0x80c7d6d20 + 0x0006e> in <filename unknown>:
0                                                                                                                                   
  at AutoMapper.Mappers.TypeMapObjectMapperRegistry+PropertyMapMappingStrategy.Map (AutoMapper.ResolutionContext context, IMappingEn
gineRunner mapper) <0x80c7cb5a0 + 0x00331> in <filename unknown>:0                                                                  
  at AutoMapper.Mappers.TypeMapMapper.Map (AutoMapper.ResolutionContext context, IMappingEngineRunner mapper) <0x80c7c82a0 + 0x001e0
> in <filename unknown>:0                                                                                                           
  at AutoMapper.MappingEngine.AutoMapper.IMappingEngineRunner.Map (AutoMapper.ResolutionContext context) <0x80c7c7430 + 0x00673> in 
<filename unknown>:0                                                                                                                
  --- End of inner exception stack trace ---                                                                                        
  at Jackett.Indexers.BaseIndexer+<FollowIfRedirect>c__async0.MoveNext () <0x80d71acc0 + 0x0026a> in <filename unknown>:0           
--- End of stack trace from previous location where exception was thrown ---                                                        
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x80d71d860 + 0x00038> in <filename unknown>:0                
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x80d71d360 + 0x00118> in <f
ilename unknown>:0                                                                                                                  
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x80d71
d1d0 + 0x000b3> in <filename unknown>:0                                                                                             
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x80897a400 + 0x00042> in <filename
 unknown>:0                                                                                                                         
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () <0x80c75c4c0 + 0x0001a> in <filename unknown>:0                       
  at Jackett.Indexers.AnimeBytes+<ApplyConfiguration>c__async0.MoveNext () <0x80c7b39c0 + 0x00d3c> in <filename unknown>:0          
--- End of stack trace from previous location where exception was thrown ---                                                        
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x80d71d860 + 0x00038> in <filename unknown>:0                
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x80d71d360 + 0x00118> in <f
ilename unknown>:0                                                                                                                  
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x80d71
d1d0 + 0x000b3> in <filename unknown>:0                                                                                             
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x80897a400 + 0x00042> in <filename
 unknown>:0                                                                                                                         
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () <0x80d71dbc0 + 0x0001a> in <filename unknown>:0            
  at Jackett.Controllers.AdminController+<Configure>c__async4.MoveNext () <0x80c7b1d10 + 0x005be> in <filename unknown>:0

**Comment by kaso17** [comment]
please also post the lines before that error

**Comment by armani316** [comment]
@AliShaikh94 Were you able to obtain the full enhanced log, including the lines @kaso17 was referring to?

**Comment by armani316** [comment]
@nmrcarroll If you're still having the same issue as @AliShaikh94, can you provide the full enhanced log, including the lines @kaso17 was referring to?

**Comment by AliShaikh94** [comment]
![image](https://cloud.githubusercontent.com/assets/7599384/22571766/38ec4d10-e96f-11e6-82a6-c4bf5f63b44e.png)


2017-02-02 17:43:17.3247 Error got NotImplemented/InternalServerError 
2017-02-02 17:43:17.3868 Error request.Method: GET 
2017-02-02 17:43:17.3874 Error request.Url: https://animebytes.tv/user/login 
2017-02-02 17:43:17.3874 Error request.Cookies:  
2017-02-02 17:43:17.3883 Error request.Referer:  
2017-02-02 17:43:17.3883 Error request.RawPOSTDdata:  
2017-02-02 17:43:17.3892 Error cookies:  
2017-02-02 17:43:17.3892 Error headerString:
 
2017-02-02 17:43:17.4957 Error got NotImplemented/InternalServerError 
2017-02-02 17:43:17.4963 Error request.Method: POST 
2017-02-02 17:43:17.4963 Error request.Url: https://animebytes.tv/user/login 
2017-02-02 17:43:17.4973 Error request.Cookies:  
2017-02-02 17:43:17.4973 Error request.Referer:  
2017-02-02 17:43:17.4982 Error request.RawPOSTDdata:  
2017-02-02 17:43:17.4982 Error cookies:  
2017-02-02 17:43:17.4991 Error headerString:
 
2017-02-02 17:43:17.5018 Error Exception in Configure AutoMapper.AutoMapperMappingException: 

Mapping types:
WebClientByteResult -> WebClientStringResult
Jackett.Utils.Clients.WebClientByteResult -> Jackett.Utils.Clients.WebClientStringResult

Destination path:
WebClientStringResult

Source value:
Jackett.Utils.Clients.WebClientByteResult ---> System.ArgumentNullException: Array cannot be null.
Parameter name: bytes
  at System.Text.Encoding.GetString (System.Byte[] bytes) <0x806359340 + 0x0008a> in <filename unknown>:0 
  at Jackett.JackettModule.<Load>m__2 (Jackett.Utils.Clients.WebClientByteResult be, Jackett.Utils.Clients.WebClientStringResult str) <0x80e800ce0 + 0x00030> in <filename unknown>:0 
  at AutoMapper.Internal.MappingExpression`2+<>c__DisplayClass58_0[TSource,TDestination].<AfterMap>b__0 (System.Object src, System.Object dest) <0x80e800c40 + 0x00094> in <filename unknown>:0 
  at AutoMapper.TypeMap.<get_AfterMap>b__40_0 (System.Object src, System.Object dest) <0x80e800b00 + 0x0006e> in <filename unknown>:0 
  at AutoMapper.Mappers.TypeMapObjectMapperRegistry+PropertyMapMappingStrategy.Map (AutoMapper.ResolutionContext context) <0x80cbf4270 + 0x00325> in <filename unknown>:0 
  at AutoMapper.MappingEngine.Map (AutoMapper.ResolutionContext context) <0x80cbf0c30 + 0x0028b> in <filename unknown>:0 
  --- End of inner exception stack trace ---
  at Jackett.Indexers.BaseIndexer+<FollowIfRedirect>c__async0.MoveNext () <0x80eb9dd00 + 0x0026a> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x80eba0910 + 0x00038> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x80eba0410 + 0x00118> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x80eba0340 + 0x000b3> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x806be8c00 + 0x00042> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.GetResult () <0x808f83f30 + 0x0001a> in <filename unknown>:0 
  at Jackett.Indexers.AnimeBytes+<ApplyConfiguration>c__async0.MoveNext () <0x80eb96930 + 0x00d48> in <filename unknown>:0 
--- End of stack trace from previous location where exception was thrown ---
  at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw () <0x80eba0910 + 0x00038> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (System.Threading.Tasks.Task task) <0x80eba0410 + 0x00118> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (System.Threading.Tasks.Task task) <0x80eba0340 + 0x000b3> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter.ValidateEnd (System.Threading.Tasks.Task task) <0x806be8c00 + 0x00042> in <filename unknown>:0 
  at System.Runtime.CompilerServices.TaskAwaiter`1[TResult].GetResult () <0x80eba0c70 + 0x0001a> in <filename unknown>:0 
  at Jackett.Controllers.AdminController+<Configure>c__async4.MoveNext () <0x80eb94cb0 + 0x005be> in <filename unknown>:0 





Here is the full log after updating to latest(0.7.983.0) again.

**Comment by nmrcarroll** [comment]
Updating to latest 0.7.983.0 has resolved the issue for me after restarting jackett and readding animebytes.

**Comment by armani316** [comment]
@AliShaikh94 Can you try to delete and and re-add AnimeBytes like @nmrcarroll did? If that does not solve the issue, let us know.

**Comment by AliShaikh94** [comment]
Still getting the same thing unfortunately. On the bright side iptorrents search seems to have been fixed with this latest release.

**Comment by kaso17** [comment]
"Error got NotImplemented/InternalServerError" typically means that libcurl can't verify the certificate. please try running jackett with "--IgnoreSslErrors true" to temporarly disable certificate validation.
Also try running "curl -v https://animebytes.tv/user/login" to see if there are any certificate validation errors.

**Comment by armani316** [comment]
@AliShaikh94 Did you manage to follow @kaso17's instructions? Did that solve the issue for you?

**Comment by AliShaikh94** [comment]
I fixed the issue. It was an SSL issue just like you said. I am running jackett on a FreeNAS box, in a jail I created back when I still had version 9.3. That version only shipped with SSL 0.9.8, and didn't support TLS1.2 properly. I had updated my FreeNAS to 9.10, and the openssl to 1.01, but it didn't update it for any of the jails. I tried updating the ssl version, but that was getting frustrating as I had to do it from the freenas port system. So I ended up just recreating the jail with the new template(with new openssl), and voila! It works. 

Thanks a bunch for your help guys!
